install/kubernetes/cilium/templates/cilium-configmap.yaml

```
@@ -885,6 +885,36 @@ data:
   unmanaged-pod-watcher-interval: "0"
 {{- end }}
 
+{{- if .Values.dnsProxy }}
+  {{- if .Values.dnsProxy.dnsRejectResponseCode }}
+  tofqdns-dns-reject-response-code: {{ .Values.dnsProxy.dnsRejectResponseCode | quote }}
+  {{- end }}
+  {{- if .Values.dnsProxy.enableDnsCompression }}
+  tofqdns-enable-dns-compression: "true"
+  {{- end }}
```

Since [the daemon default value is `true`](https://github.com/cilium/cilium/blob/5ba662563e273658086052c27028fa459d84de0e/pkg/defaults/defaults.go#L138-L140), there is now way to disable it from Helm as `--set dnsProxy.enableDnsCompression=false` will omit the option from the ConfigMap and use the daemon's default. Is that a fair statement?
```suggestion
  tofqdns-enable-dns-compression: {{ .Values.dnsProxy.enableDnsCompression | quote }}
```

---

Good catch @kaworu. PR updated. 
For consistency, I've kept the value check before inserting it into the `ConfigMap`
```
{{- if hasKey .Values.dnsProxy "enableDnsCompression" }}
  tofqdns-enable-dns-compression: {{ .Values.dnsProxy.enableDnsCompression | quote }}
{{- end }}```

---

That's reasonable, thanks for the fix!